### PR TITLE
MDS-3173 - GIS Spatial Files + MDS-3466 - Editing FAP Throws Error 

### PR DESCRIPTION
**Pull request by nayr974**
This PR addresses two tickets, one is providing some small extensions of our existing APIs for the GIS team as well as a new role for them to call our APIs with, and the other is a fix for a bug when saving edits to the FAP.

### CODE REVIEW

File: services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js [hunk 1]
```diff
@@ -326,10 +328,12 @@ export class NOWPermitGeneration extends Component {
         const fieldIdParts = key.split(/_(.+)/);
         const guid = fieldIdParts[0];
         const fieldName = fieldIdParts[1];
-        if (!(guid in allFileMetadata)) {
-          allFileMetadata[guid] = {};
+        if (guid && guid !== "null") {
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:331]: So this doesn't add the doc in the request if the guid is null, but isn't it a problem that the guid is null in the first place?

Comment (nayr974) [reply services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:331]: It is, and this is a band-aid fix for sure. I didn't find the root cause as to why it's happening and got stuck at "redux forms magic".

Comment (exzizt) [reply services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:331]: What if this causes a different more worse problem? :( Idk I haven't wrapped my head around it yet. Might be good to get someone else to look at it and get their opinion and maybe have a team discussion? @mhuseinov @mmumby

Comment (exzizt) [reply services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:331]: *Resolved in convo with team*

**Review by exzizt — APPROVED** [review]
image

**Pull request merged**